odxtools/physicaltype.py

```diff
@@ -56,21 +57,21 @@ def __post_init__(self):
             self.display_radix = Radix(self.display_radix)
 
     @staticmethod
-    def from_et(et_element, doc_frags: List[OdxDocFragment]):
-        base_data_type = et_element.get("BASE-DATA-TYPE")
-        odxassert(base_data_type in [
-            "A_INT32",
-            "A_UINT32",
-            "A_FLOAT32",
-            "A_FLOAT64",
-            "A_ASCIISTRING",
-            "A_UTF8STRING",
-            "A_UNICODE2STRING",
-            "A_BYTEFIELD",
-        ])
-        display_radix = et_element.get("DISPLAY-RADIX")
-        precision = et_element.findtext("PRECISION")
-        if precision is not None:
-            precision = int(precision)
+    def from_et(et_element: ElementTree.Element, doc_frags: List[OdxDocFragment]):
+        base_data_type_str = et_element.get("BASE-DATA-TYPE")
+        if base_data_type_str not in DataType.__members__:
+            odxraise(f"Encountered unknown base data type '{base_data_type_str}'")
+        base_data_type = DataType(base_data_type_str)
+
+        display_radix_str = et_element.get("DISPLAY-RADIX")
+        if display_radix_str is not None:
+            if display_radix_str not in Radix.__members__:
```

Comment:
It would help more if radix was an int, 10 for decimal, 16 for hex.. etc

Reply:
We could make it so that the enum is an IntEnum, that won't be breaking the spec

Reply:
how to create this object from a string? i.e. something like this, but working:
```python
class Radix(IntEnum):
  DEC = 10
  HEX = 16

r = Radix("HEX")
```

Reply:
okay, found a solution: 9387b34e49cb35176811b3e26654fa7bad1da866 . (thanks for your persistence)

note though, that this solution only works as long as all members are valid python identifiers, in particular this approach won't work if the names contain a minus...